### DISCUSSION TITLE
Engine Debug Slowly

### DISCUSSION THREAD

**Discussion by lzh-maker** [OpenRadioss/OpenRadioss#758] [Q&A]
Hello all,
I have compiled and  debugged the engine with virtual machine Vmware and ubuntu or Centos,two types Linux systems  have the same performance,the speed of debugging is very slow,compared to ubuntu,Centos  perform  a litter faster ,why?

**Comment by PaulAltair** [comment]
same gcc? Latest ubuntu ships with later gcc I think?

**Reply by lzh-maker** [reply]
You mean to use gcc to compile the code instead of using the official compilation tool？I don't know if it's a compiler problem or a debugging tool problem。And have you used any tools to debug, I used vscode with fortran plugin before, the debugging is very slow。

**Reply by PaulAltair** [reply]
Sorry, I'm almost completely new to compiling and running source code, so I might well have been asking something stupid! 

I've never used a debugger and not sure what you mean by 'official compilation tool'? I just build with the supplied build.sh scripts in engine and starter source folders (per the getting started guides) but they use gcc I believe.

I made the suggestion re: gcc version as I was helping someone with an OpenRadioss run issue a few weeks ago, they were using a RH based distro and I was using Ubuntu and not getting the same issue.

In the end  we found the difference was in a bug exposed by code compiling between gcc10 and gcc8 in the 2 releases, the latest Ubuntu LTS releases have gcc10 so I thought it might possibly explain a difference in performance in your case if your centos is using gcc8 (on an otherwise similar system).

**Reply by lzh-maker** [reply]
I'll try to find if it's a gcc version problem, but I don't think it's likely，because I'm using a newer version

**Comment by OlivierWienholtz** [comment]
Hello,

Sorry for late answer.
Debug version is compiled with O0 + aggressive debug flags to see most of issues. 
In fact variables or some loops in source code can be optimized away with the Optimization.
This permits to see better issues related to source code.

As a side effect : as all optimizations are turned off the OpenRadioss is much slower.

**Reply by lzh-maker** [reply]
Thank you for your reply, I am a beginner, can you tell me how to enable code optimization，for details

**Comment by OlivierWienholtz** [comment]
It is possible to have full Optimization + generic debug flags : 
Use the -addflag="-g -fbacktrace"
This will those to the set of Compilation flags.

./build_script.sh -arch=linux64_gf -addflag="-g -fbacktrace"
